src/main/java/org/jabref/logic/cleanup/ConvertToBiblatexCleanup.java

```diff
@@ -39,8 +41,26 @@
                 entry.clearField(StandardField.YEAR).ifPresent(changes::add);
                 entry.clearField(StandardField.MONTH).ifPresent(changes::add);
             });
+        } else {
+            // If the year from date field is filled and equal to year it should be removed the year field
+            entry.getFieldOrAlias(StandardField.DATE).ifPresent(date -> {
+                entry.getFieldOrAlias(StandardField.YEAR).ifPresent(year -> {
```

Comment:
One of the Date.parse method overloads accepts already Optional as parameter, so you can make your code a bit easier and remove the ifPressent checks for Year and month

so you would have` checkDate = Date.parse(entry.getFieldOrAlias(StandardField.YEAR),  entry.getFieldOrAlias(StandardField.Month), Optional.empty())`

Reply:
@Siedlerchr Thanks for the tip. Already did that.